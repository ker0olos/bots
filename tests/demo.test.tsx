// deno-lint-ignore-file no-explicit-any

import React from 'https://esm.sh/react@18.2.0';

import { renderToString } from 'https://esm.sh/react-dom@18.2.0/server';

import { FakeTime } from 'https://deno.land/std@0.179.0/testing/time.ts';

import { assertSnapshot } from 'https://deno.land/std@0.179.0/testing/snapshot.ts';

import {
  App,
  Desc,
  Embed,
  Gacha,
  Image,
  Media,
  Rating as RatingElement,
  Title,
} from '../src/demo.tsx';

import type { Pull } from '../src/gacha.ts';

import Rating from '../src/rating.ts';

Deno.test('<Embed/>', async (test) => {
  await assertSnapshot(
    test,
    renderToString(
      <Embed>
        <p>paragraph</p>
      </Embed>,
    ),
  );
});

Deno.test('<Title/>', async (test) => {
  await assertSnapshot(
    test,
    renderToString(
      <Title text={'title'} />,
    ),
  );
});

Deno.test('<Desc/>', async (test) => {
  await assertSnapshot(
    test,
    renderToString(
      <Desc text={'description'} />,
    ),
  );
});

Deno.test('<Image/>', async (test) => {
  await assertSnapshot(
    test,
    renderToString(
      <Image src={'src'} />,
    ),
  );
});

Deno.test('<Media/>', async (test) => {
  const pull = {
    media: {
      title: {
        english: 'title',
      },
      images: [{
        url: 'url',
      }],
    },
  };

  const timeStub = new FakeTime();

  try {
    await assertSnapshot(
      test,
      renderToString(
        <Media pull={pull as any} />,
      ),
    );

    await timeStub.tickAsync(4000);
  } finally {
    timeStub.restore();
  }
});

Deno.test('<Rating/>', async (test) => {
  const pull = {
    rating: new Rating({ stars: 3 }),
  };

  const timeStub = new FakeTime();

  try {
    await assertSnapshot(
      test,
      renderToString(
        <RatingElement pull={pull as any} />,
      ),
    );

    await timeStub.tickAsync(5000);
  } finally {
    timeStub.restore();
  }
});

Deno.test('<Gacha/>', async (test) => {
  const pull = {
    rating: new Rating({ stars: 3 }),
    character: {
      name: {
        english: 'name',
      },
      images: [{
        url: 'url',
      }],
    },
    media: {
      title: {
        english: 'title',
      },
    },
  };

  await assertSnapshot(
    test,
    renderToString(
      <Gacha pull={pull as Pull} />,
    ),
  );
});

Deno.test('<App/>', async (test) => {
  await assertSnapshot(
    test,
    renderToString(
      <App>
        <p>paragraph</p>
      </App>,
    ),
  );
});

Deno.test('<App/> Suspended', async (test) => {
  const T = () => {
    return {
      read(): any {
        switch (false) {
          default:
            // deno-lint-ignore no-throw-literal
            throw '';
        }
      },
    };
  };

  const TT = () => <img src={T().read()} />;

  await assertSnapshot(
    test,
    renderToString(
      <App>
        <TT />
      </App>,
    ),
  );
});
