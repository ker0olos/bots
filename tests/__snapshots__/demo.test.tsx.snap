export const snapshot = {};

snapshot[`<Embed/> 1`] = `'<div style="overflow:hidden;background-color:#2b2d31;padding:.5rem 1rem 1rem .75rem;border-left:4px solid #1e1f22;border-radius:4px;line-height:22px"><p>paragraph</p></div>'`;

snapshot[`<Title/> 1`] = `'<div style="white-space:break-spaces;margin-top:8px;margin-bottom:2px;line-height:1.125rem;color:#e0e1e5;font-size:0.875rem;font-family:&#x27;Noto Sans&#x27;, sans-serif;font-weight:600">title</div>'`;

snapshot[`<Desc/> 1`] = `'<div style="white-space:break-spaces;line-height:1.125rem;color:#e0e1e5;font-size:0.875rem;font-family:&#x27;Noto Sans&#x27;, sans-serif;font-weight:700">description</div>'`;

snapshot[`<Image/> 1`] = `'<img style="width:auto;min-width:213px;min-height:300px;max-height:300px;border-radius:4px;margin-top:16px" src="src"/>'`;

snapshot[`<Media/> 1`] = `'<!--\$!--><template></template><div style="white-space:break-spaces;margin-top:8px;margin-bottom:2px;line-height:1.125rem;color:#e0e1e5;font-size:0.875rem;font-family:&#x27;Noto Sans&#x27;, sans-serif;font-weight:600">title</div><img style="width:auto;min-width:213px;min-height:300px;max-height:300px;border-radius:4px;margin-top:16px" src="/external/url"/><!--/\$-->'`;

snapshot[`<Rating/> 1`] = `'<!--\$!--><template></template><img style="width:auto;min-width:213px;min-height:300px;max-height:300px;border-radius:4px;margin-top:16px" src="/assets/stars/3.gif"/><!--/\$-->'`;

snapshot[`<Gacha/> 1`] = `'<div style="display:flex;margin-top:8px"><img style="width:18px;height:18px" src="https://cdn.discordapp.com/emojis/1061016362832642098.webp?size=44&amp;quality=lossless"/><img style="width:18px;height:18px" src="https://cdn.discordapp.com/emojis/1061016362832642098.webp?size=44&amp;quality=lossless"/><img style="width:18px;height:18px" src="https://cdn.discordapp.com/emojis/1061016362832642098.webp?size=44&amp;quality=lossless"/><img style="width:18px;height:18px" src="https://cdn.discordapp.com/emojis/1061016360190222466.webp?size=44&amp;quality=lossless"/><img style="width:18px;height:18px" src="https://cdn.discordapp.com/emojis/1061016360190222466.webp?size=44&amp;quality=lossless"/></div><div style="white-space:break-spaces;margin-top:8px;margin-bottom:2px;line-height:1.125rem;color:#e0e1e5;font-size:0.875rem;font-family:&#x27;Noto Sans&#x27;, sans-serif;font-weight:600">title</div><div style="white-space:break-spaces;line-height:1.125rem;color:#e0e1e5;font-size:0.875rem;font-family:&#x27;Noto Sans&#x27;, sans-serif;font-weight:700">name</div><img style="width:auto;min-width:213px;min-height:300px;max-height:300px;border-radius:4px;margin-top:16px" src="/external/url"/>'`;

snapshot[`<App/> 1`] = `'<html><head><title>Fable</title><link rel="icon" type="image/x-icon" href="https://raw.githubusercontent.com/ker0olos/fable/main/assets/icon.png"/><link rel="preconnect" href="https://fonts.googleapis.com"/><link rel="preconnect" crossorigin="anonymous" href="https://fonts.gstatic.com"/><link href="https://fonts.googleapis.com/css2?family=Noto+Sans:wght@400;600;700&amp;display=swap" rel="stylesheet"/></head><body style="display:flex;background-color:#313338;justify-content:center;align-items:center;min-height:100vh;padding:0;margin:0"><div style="display:flex;flex-direction:column;gap:8px"><div style="overflow:hidden;background-color:#2b2d31;padding:.5rem 1rem 1rem .75rem;border-left:4px solid #1e1f22;border-radius:4px;line-height:22px"><!--\$--><p>paragraph</p><!--/\$--></div><div style="display:flex;gap:6px"><a style="display:flex;cursor:pointer;color:white;background-color:#5865f2;justify-content:center;align-items:center;padding:2px 16px;min-height:38px;border:0;border-radius:3px;font-size:0.875rem;font-family:&#x27;Noto Sans&#x27;, sans-serif;text-decoration:none;font-weight:400" href="https://discord.com/api/oauth2/authorize?client_id=undefined&amp;scope=applications.commands">Add Fable to Your Server</a><a style="display:flex;cursor:pointer;color:white;background-color:#4e5058;justify-content:center;align-items:center;padding:2px 16px;min-height:38px;border:0;border-radius:3px;font-size:0.875rem;font-family:&#x27;Noto Sans&#x27;, sans-serif;text-decoration:none;font-weight:400" href="/demo">Again</a></div><a style="display:flex;cursor:pointer;color:white;background-color:#4e5058;justify-content:center;align-items:center;padding:2px 16px;min-height:38px;border:0;border-radius:3px;font-size:0.875rem;font-family:&#x27;Noto Sans&#x27;, sans-serif;text-decoration:none;font-weight:400" href="https://github.com/ker0olos/fable">GitHub</a></div></body></html>'`;

snapshot[`<App/> Suspended 1`] = `'<html><head><title>Fable</title><link rel="icon" type="image/x-icon" href="https://raw.githubusercontent.com/ker0olos/fable/main/assets/icon.png"/><link rel="preconnect" href="https://fonts.googleapis.com"/><link rel="preconnect" crossorigin="anonymous" href="https://fonts.gstatic.com"/><link href="https://fonts.googleapis.com/css2?family=Noto+Sans:wght@400;600;700&amp;display=swap" rel="stylesheet"/></head><body style="display:flex;background-color:#313338;justify-content:center;align-items:center;min-height:100vh;padding:0;margin:0"><div style="display:flex;flex-direction:column;gap:8px"><div style="overflow:hidden;background-color:#2b2d31;padding:.5rem 1rem 1rem .75rem;border-left:4px solid #1e1f22;border-radius:4px;line-height:22px"><!--\$!--><template></template><img style="width:auto;min-width:213px;min-height:300px;max-height:300px;border-radius:4px;margin-top:16px" src="/assets/spinner.gif"/><!--/\$--></div><div style="display:flex;gap:6px"><a style="display:flex;cursor:pointer;color:white;background-color:#5865f2;justify-content:center;align-items:center;padding:2px 16px;min-height:38px;border:0;border-radius:3px;font-size:0.875rem;font-family:&#x27;Noto Sans&#x27;, sans-serif;text-decoration:none;font-weight:400" href="https://discord.com/api/oauth2/authorize?client_id=undefined&amp;scope=applications.commands">Add Fable to Your Server</a><a style="display:flex;cursor:pointer;color:white;background-color:#4e5058;justify-content:center;align-items:center;padding:2px 16px;min-height:38px;border:0;border-radius:3px;font-size:0.875rem;font-family:&#x27;Noto Sans&#x27;, sans-serif;text-decoration:none;font-weight:400" href="/demo">Again</a></div><a style="display:flex;cursor:pointer;color:white;background-color:#4e5058;justify-content:center;align-items:center;padding:2px 16px;min-height:38px;border:0;border-radius:3px;font-size:0.875rem;font-family:&#x27;Noto Sans&#x27;, sans-serif;text-decoration:none;font-weight:400" href="https://github.com/ker0olos/fable">GitHub</a></div></body></html>'`;
